utils/build-script-impl

```diff
@@ -409,58 +443,31 @@ function set_build_options_for_host() {
     local host="$1"
 
     # Hosts which can be cross-compiled must specify:
-    # SWIFT_HOST_TRIPLE and llvm_target_arch (as well as usual HOST_VARIANT flags)
+    # SWIFT_HOST_TRIPLE and llvm_target_arch.
+    # Hosts which have differing platforn names from their
+    # SWIFT_HOST_VARIANT_* values should change them here as well.
+
+    verify_host_is_supported $host
+
+    local IFS='-'
+    local plat_and_arch=($host)
+    local platform=${plat_and_arch[0]}
+    local architecture=${plat_and_arch[1]}
+    local IFS=' '
+
+    SWIFT_HOST_VARIANT=$platform
+    SWIFT_HOST_VARIANT_SDK=$(toupper $platform)
+    SWIFT_HOST_VARIANT_ARCH=$architecture
 
     case ${host} in
-        freebsd-x86_64)
-            SWIFT_HOST_VARIANT="freebsd"
-            SWIFT_HOST_VARIANT_SDK="FREEBSD"
-            SWIFT_HOST_VARIANT_ARCH="x86_64"
-            ;;
-        cygwin-x86_64)
-            SWIFT_HOST_VARIANT="cygwin"
-            SWIFT_HOST_VARIANT_SDK="CYGWIN"
-            SWIFT_HOST_VARIANT_ARCH="x86_64"
+        linux-armv6)
+            SWIFT_HOST_TRIPLE="armv6-unknown-linux-gnueabihf"
+            llvm_target_arch="ARM"
             ;;
-        haiku-x86_64)
-            SWIFT_HOST_VARIANT="haiku"
-            SWIFT_HOST_VARIANT_SDK="HAIKU"
-            SWIFT_HOST_VARIANT_ARCH="x86_64"
+        linux-armv7)
+            SWIFT_HOST_TRIPLE="armv7-unknown-linux-gnueabihf"
```

Comment:
This seems ... limiting.  Not all ARMv7 targets are HF.

Reply:
This just preserves existing behavior, I don't know why this is the way it is.